WORKSPACE

```diff
@@ -594,3 +598,31 @@ debian_deps()
 load("@bazel_skylib//:workspace.bzl", "bazel_skylib_workspace")
 
 bazel_skylib_workspace()
+
+load("@rules_jvm_external//:repositories.bzl", "rules_jvm_external_deps")
+
+rules_jvm_external_deps()
+
+load("@rules_jvm_external//:setup.bzl", "rules_jvm_external_setup")
+
+rules_jvm_external_setup()
+
+load("@rules_jvm_external//:defs.bzl", "maven_install")
+
+maven_install(
```

Comment:
Please also use [pinning](https://github.com/bazelbuild/rules_jvm_external#pinning-artifacts-and-integration-with-bazels-downloader) to use bazel's downloader and repo caching instead of coursier.

Reply:
Oh, I see, thanks! Does the pinning also work with Bzlmod?

Reply:
It should! @shs96c could correct me if it doesn't :)

Reply:
Nice, I'll try it!

Reply:
Done, please take another look ;)

Reply:
Looks like the vendoring of `@maven` no longer works, `glob(["v1/**/*.jar"])` doesn't work anymore since jar files are copied by a genrule from their own repo. Have to figure out a workaround..

Reply:
https://github.com/bazelbuild/rules_jvm_external#accessing-transitive-dependencies-list may be useful to generate BUILD rules on the Bazel side if all you need is a way to get the complete list of artifacts.

Reply:
OK, I still have to hack rules_jvm_external to make it work. Basically, I need to mock the maven repository for the bootstrap build, which includes vendoring all the jar files and generate a usable BUILD file.